### ISSUE TITLE
Format Ruby

### ISSUE THREAD

**Issue by erbridge** [geeksforsocialchange/PlaceCal#1456]
## User story

As a developer I need to have my Ruby code formatted by a computer so that I know it follows the repository code standards and doesn't have make formatting choices.

## Acceptance criteria

- [ ] Ruby code is formatted while committing

## Implementation plan

_To be written by the developer_

**Comment by aaaaargZombies** [comment]
https://github.com/adamzapasnik/prettier-plugin-erb - might be useful

**Comment by r-ferrier** [comment]
Thios has been a trip! Prettier for ruby works really well but it is not in accordance with rubocop. I changed some of the rules but I can't get the two to agree on indentation. 

Prettier wants this:
![Screenshot 2024-02-14 at 17 17 08](https://github.com/geeksforsocialchange/PlaceCal/assets/46496391/f021437b-ef16-4bb5-b9c0-e34f26cdc9cf)

Rubocop wants this:
![Screenshot 2024-02-14 at 17 16 58](https://github.com/geeksforsocialchange/PlaceCal/assets/46496391/8f604b89-673c-4a27-a3f7-76bf9b940f88)

Prettier's suggestion is really odd to me and I'm unable to work out what phrase to search for to find out why it would be doing this - the only kind of indentation rules I'm finding are two do with the age old 2 vs 4 spaces battle, nothing about indenting to the starting position of the opening bracket.

Given the above I think I might use it for a one off file format, and then run the files through rubocop? This would at least tidy some things up, whilst stopping the two from clashing. What do we think?

**Comment by r-ferrier** [comment]
discussed in tech standup - can we just run this on erb files?

**Comment by r-ferrier** [comment]
unfortunately the erb arm of this plugin is currently very broken. I've just tried to add it and ran into several problems, then found this: 

https://github.com/adamzapasnik/prettier-plugin-erb/issues/137

I then tried to find another erb editor and discovered erb-lint, which runs on rubocop - https://github.com/Shopify/erb-lint

BUT setting this up has been quite painful. It requires a lot of special config rules to stop rubocop from doing mad things to your erb, because it treats every block as a separate file, for example injecting "frozen string literal" into files thousands of times! I've tried adding config rules to address this (https://github.com/Shopify/erb-lint/issues/222) but it's still introducing lots of changes that just break the erb, like reformatting variable assignation, for example

```
<% build = ENV['GIT_REV'] ? ENV['GIT_REV'][0,7] : 'main' %>
```
becomes
```
<% ENV['GIT_REV'] ? ENV['GIT_REV'][0, 7] : 'main' %>
```

which is extremely not helpful! The main thing it does everywhere in spades is reformatting double quotations to single quotations, and it doesn't do any helpful long line or weird space reformatting, although it does tell you about long lines.

So, the summary is, this is more work than its worth to introduce and I'm going to close this now as being thoroughly researched.
